plugins/http_plugin/include/eosio/http_plugin/macros.hpp

```diff
@@ -15,7 +15,7 @@ struct async_result_visitor : public fc::visitor<fc::variant> {
          auto params = parse_params<api_namespace::call_name ## _params, params_type>(body);\
          FC_CHECK_DEADLINE(deadline);\
          api_handle.call_name( std::move(params), \
-            [cb, body](const std::variant<fc::exception_ptr, call_result>& result){\
+               [cb=std::move(cb), body=std::move(body)](const std::variant<fc::exception_ptr, call_result>& result){ \
```

Comment:
`cb` and `body` are used in the catch.

Reply:
Hum, my understanding is that the lambda we construct (into which I mode `cb` and `body`) is passed to the API as `next`, and if the API fails for any reason it invokes the `next()` function and doesn't rethrow. That is why we check for exceptions inside the lambda.

So either we construct the lambda, and we pass it to the API, or there was an exception which happened *before* the API was called in which case the bottom catc clause is used.